src/AddOn.Episerver.Settings/Core/SettingsService.cs

```diff
@@ -243,29 +304,47 @@ public IEnumerable<ContentReference> SettingsRoots
     ///     The <see cref="T:System.Threading.ReaderWriterLockSlim" /> object
     ///     has been disposed.
     /// </exception>
-    public T GetSettings<T>(IContent content)
-            where T : SettingsBase
+    public T GetSetting<T>(IContent content) where T : SettingsBase
+    {
+        return GetSettingsRecursive<T>(content).FirstOrDefault();
```

Comment:
From a performance perspective it feels unnecessary to fetch all settings and then just pick the first. I understand that it's good to reuse the logic, perhaps we can have an optional count parameter to the method call and check this to abort the recursion if we exceed this?

Reply:
We don't get all the settings since it's using an iterator and yield. It will stop "pumping" for more values as soon as a first value has arrived. Pretty neat if you ask me :-)

Reply:
Ahh, thanks for informing me - I've learnt something new :)